collector/cluster_health.go

```diff
@@ -0,0 +1,246 @@
+package collector
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	namespace = "elasticsearch"
+)
+
+type ClusterHealth struct {
+	logger log.Logger
+	client *http.Client
+	url    *url.URL
+
+	ActivePrimaryShards     *prometheus.Desc
+	ActiveShards            *prometheus.Desc
+	DelayedUnassignedShards *prometheus.Desc
+	InitializingShards      *prometheus.Desc
+	NumberOfDataNodes       *prometheus.Desc
+	NumberOfInFlightFetch   *prometheus.Desc
+	NumberOfNodes           *prometheus.Desc
+	NumberOfPendingTasks    *prometheus.Desc
+	RelocatingShards        *prometheus.Desc
+	StatusIsGreen           *prometheus.Desc
+	Status                  *prometheus.Desc
+	StatusIsYellow          *prometheus.Desc
+	StatusIsRed             *prometheus.Desc
+	TimedOut                *prometheus.Desc
+	UnassignedShards        *prometheus.Desc
+}
+
+func NewClusterHealth(logger log.Logger, client *http.Client, url *url.URL) *ClusterHealth {
+	subsystem := "cluster_health"
+
+	return &ClusterHealth{
+		logger: logger,
+		client: client,
+		url:    url,
+
+		ActivePrimaryShards: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, subsystem, "active_primary_shards"),
+			"Tthe number of primary shards in your cluster. This is an aggregate total across all indices.",
+			[]string{"cluster"}, nil,
+		),
+		ActiveShards: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, subsystem, "active_shards"),
+			"Aggregate total of all shards across all indices, which includes replica shards.",
+			[]string{"cluster"}, nil,
+		),
+		DelayedUnassignedShards: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, subsystem, "delayed_unassigned_shards"),
+			"XXX WHAT DOES THIS MEAN?",
+			[]string{"cluster"}, nil,
+		),
+		InitializingShards: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, subsystem, "initializing_shards"),
+			"Count of shards that are being freshly created.",
+			[]string{"cluster"}, nil,
+		),
+		NumberOfDataNodes: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, subsystem, "number_of_data_nodes"),
+			"Number of data nodes in the cluster.",
+			[]string{"cluster"}, nil,
+		),
+		NumberOfInFlightFetch: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, subsystem, "number_of_in_flight_fetch"),
+			"The number of ongoing shard info requests.",
+			[]string{"cluster"}, nil,
+		),
+		NumberOfNodes: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, subsystem, "number_of_nodes"),
+			"Number of nodes in the cluster.",
+			[]string{"cluster"}, nil,
+		),
+		NumberOfPendingTasks: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, subsystem, "number_of_pending_tasks"),
+			"XXX WHAT DOES THIS MEAN?",
```

Comment:
Intended or left over? If we can't figure out a proper description we may want to put an `undocumented` instead.